### PR TITLE
suppermatter doesn't get angry with mindless wights

### LINKED ISSUE #55959: SM Infinite Radiation Bug
## Round ID: 
Round ID: [153637](https://scrubby.melonmesa.com/round/153637)

## Reproduction:
The Supermatter crystal was broken in such a way that it threw off infinite radiation without ever exploding at 0%. Please take a look at the moment when Carl Kofstri and Kevin Calf are setting up the SM -- the valves are activated and the air alarm is configured. Those are the only steps taken and the SM proceeds to generate radiation infinitely.

### PR DESCRIPTION
**Pull request by spessbro**
## About The Pull Request
stops the sm from getting upset when you put a brainless man in front of it

Fixes #55959

## Why It's Good For The Game
i need a fix, stat!

## Changelog
:cl:
fix: Fixed SM not delamming if a brainless human was near it.
/:cl:

### CODE REVIEW

**Comment by Jared-Fogle** [comment]
Please consider this a friendly warning, your PR's must have a properly descriptive title and body content that explains the purpose of the PR.

Your current title and or body content does not meet requirements, so please update it.

**Review by LemonInTheDark — DISMISSED** [review]
Minorly changes behavior, but it really doesn't matter

**Comment by Qustinnus** [comment]
does it?

**Comment by LemonInTheDark** [comment]
it means living beings without a mind won't have their hallucination var increase

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -709,6 +709,8 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 	for(var/mob/living/carbon/human/l in view(src, HALLUCINATION_RANGE(power)))
 		// Someone (generally a Psychologist), when looking at the SM
 		// within hallucination range makes it easier to manage.
+		if (isnull(l.mind))
+    		continue
```
Comment (LemonInTheDark) [inline code/modules/power/supermatter/supermatter.dm:713]: Are you using spacing here or something?

File: code/modules/power/supermatter/supermatter.dm [hunk 2]
```diff
@@ -709,6 +709,8 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 	for(var/mob/living/carbon/human/l in view(src, HALLUCINATION_RANGE(power)))
 		// Someone (generally a Psychologist), when looking at the SM
 		// within hallucination range makes it easier to manage.
+		if (isnull(l.mind))
+			continue
 		if(HAS_TRAIT(l.mind, TRAIT_SUPERMATTER_SOOTHER))
```
Comment (TemporalOroboros) [inline code/modules/power/supermatter/supermatter.dm:714]: Wouldn't appending the mind check to the front of this if statement work just as well?

Comment (Jared-Fogle) [reply code/modules/power/supermatter/supermatter.dm:714]: no, because its checked again later. easier to just check in the beginning

**Comment by oklews** [comment]
Feature

**Review by LemonInTheDark — DISMISSED** [review]

File: code/modules/power/supermatter/supermatter.dm [hunk 3]
```diff
@@ -709,6 +709,8 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 	for(var/mob/living/carbon/human/l in view(src, HALLUCINATION_RANGE(power)))
 		// Someone (generally a Psychologist), when looking at the SM
 		// within hallucination range makes it easier to manage.
+		if (isnull(l.mind))
```
Comment (Ghommie) [inline code/modules/power/supermatter/supermatter.dm:712]:
```suggestion
		if (QDELETED(l.mind))
```

**Pull request merged**